### PR TITLE
feat: CI Pipeline to build and test CPP SDK core module

### PR DESCRIPTION
**Pull request by AdityaKasar**

### CODE REVIEW

File: .github/workflows/cxx-build.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: CXX build
+on:
+  workflow_dispatch:
+  push:
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:4]: We would want these running on PRs, not pushes.

File: .github/workflows/utils.sh [hunk 2]
```diff
@@ -152,15 +154,214 @@ function unzipArtifact(){
...
+
+  cd src/sdks/core
+  ls -la build/cpp/src
+  cp build/cpp/src/firebolt-core-native-sdk-${FIREBOLT_VERSION}.tgz $current_dir/data
+  cd $current_dir/data
+  tar -zxvf firebolt-core-native-sdk-${FIREBOLT_VERSION}.tgz
+  cd firebolt-core-native-sdk-${FIREBOLT_VERSION}/
+  chmod +x ./build.sh
+  sed -i -e 's/prefix=/prefix /g' build.sh
+  cat ./build.sh
```
Comment (kschrief) [inline .github/workflows/utils.sh:289]: Why are we "cat"ing this file and others below?

File: .github/workflows/utils.sh [hunk 3]
```diff
@@ -152,15 +154,214 @@ function unzipArtifact(){
...
+  -DPERSISTENT_PATH="${PWD}/install/var/wpeframework" \
+  -DPORT="55555" \
+  -DPROXYSTUB_PATH="${PWD}/install/usr/lib/wpeframework/proxystubs" \
+  -DSYSTEM_PATH="${PWD}/install/usr/lib/wpeframework/plugins" \
+  -DVOLATILE_PATH="tmp" && echo "Thunder configure succeeded" || exit 9999
+
+  cmake --build build/Thunder --target install && echo "Thunder Build succeeded" || exit 9999
+}
+
+function buildCoreCPPSDK() {
```
Comment (kschrief) [inline .github/workflows/utils.sh:275]: Can we combine the 3 build functions into one function that takes in an SDK parameter? We'll have other SDKs to worry about down the road and it'll be easier to add it as a parameter.

File: .github/workflows/cxx-build.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        run: |
+          npm run dist
+
+      - name: Build CXX 
+        run: |
+          cd src/sdks/core/
+          ls
+          npm run cpp
+
+      - name: Build SDK
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:50]: It looks like we're only building core here? Can we also build the other SDKs?

File: .github/workflows/utils.sh (hunk 3 above)
Comment (kschrief) [reply .github/workflows/utils.sh:275]: I also don't see these functions being used anywhere.

**Review by kschrief — CHANGES_REQUESTED** [review]

**Comment by AdityaKasar** [comment]
Requesting a re-review 

- Addressed previous comments 
- Implemented caching mechanism for faster installation of npm dependencies 
- Pre-compiling Thunder and openrpc artifacts and used artifacts uploading feature 
- Building 3 SDKs parallelly 

This is the final pipeline for reference: https://github.com/rdkcentral/firebolt-apis/actions/runs/11291931107

**Comment by AdityaKasar** [comment]
latest pipeline run: https://github.com/rdkcentral/firebolt-apis/actions/runs/11354300763/job/31581481906

**Review by kschrief — APPROVED** [review]

**Review by kschrief — APPROVED** [review]

**Pull request merged**